### PR TITLE
Project 1: Add classes

### PR DESCRIPTION
**Pull request by ookino**
### Project requirements
- Create class `Person` with  the following:
 - [x] Instance vars: `@id`, `@name`, and `@age`.
 - [x]   - Constructor with `name`,  `age`, and `parent_permission` as parameter. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
 - [x]   - Getters for `@id`, `@name`, and `@age`.
 - [x]   - Setters for `@name` and `@age`.
 - [x]   - Private method `is_of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
 - [x]   - Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.
- Create class `Student` with the following:
 - [x]   - Inherits from `Person`.
 - [x]   - Constructor extends parent's constructar by adding `@classroom` and a parameter for it.
 - [x]   - Method `play_hooky` that returns `"¯\(ツ)/¯"`.
-  Create class `Teacher` with the following:
 - [x]   - Inherits from `Person`.
 - [x]   - Constructor extends parent's constructar by adding `@specialization` and a parameter for it.
 - [x]   - Override `can_use_services?` so it always returns `true`.
 - [x] - Each class should be saved in a separate file.

### CODE REVIEW

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Yaseer, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

The Rspec linter you added to your project is failing, please fix (or remove, because it's optional) it before requesting a new review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Person
+  def initialize(id, age, name = 'unknown', parent_permission: true)
+    @id = id
```
Comment (mikethreels) [inline person.rb]: - [x] the `id` is not going to be passed as an argument, it should be assigned by the program, a good way to solve this is assigning a random number to the instance variable `@id`.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Student < Person
+  def initialize(_clasroom, age, name, parent_permission)
```
Comment (mikethreels) [inline student.rb]: - [x] in this class the `name` and `parent_permission` should be assigned a default value as well.

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Teacher < Person
+  def initialize(_specialization, age, name, parent_permission)
```
Comment (mikethreels) [inline teacher.rb]: - [x] in this class the `name` and `parent_permission` should be assigned a default value as well.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Yaseer, 

Good job setting up the different classes and the attribute accessors 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-labana — APPROVED** [review]
Hi @yaseerokino, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**